packages/astro/src/jsx/babel.ts

```diff
@@ -205,7 +205,8 @@ export default function astroJSX(): PluginObj {
 							break;
 						}
 						if (namespace.at(0) === local) {
-							path.setData('import', { name: imported, path: source });
+							const name = imported === '*' ? imported : tagName;
```

Comment:
Components that use an `ImportNamespaceSpecifier` will have their import `name` set to `'*'`, which is handled here:
https://github.com/withastro/astro/blob/dfa3519d4c1aba888dceec50f285677d659d1540/packages/astro/src/jsx/babel.ts#L71-L73

In other cases, we'll just use the tag name, which will be resolved in the same way that the namespace is. That happens here:
https://github.com/withastro/astro/blob/dfa3519d4c1aba888dceec50f285677d659d1540/packages/astro/src/runtime/server/astro-island.ts#L72-L75